tests/conftest.py

```diff
@@ -204,3 +205,112 @@ def _R(expr):
         assert len(base) == 1
         base = base.pop()
     return EvalDerivative(*expr.args, base=base)
+
+
+# Utilities for testing tree structure
+
+
+def assert_structure(operator, exp_trees=None, exp_iters=None):
+    """
+    Utility function that helps to check loop structure of IETs. Retrieves trees from an
+    Operator and check that the blocking structure is as expected. Trees and Iterations
+    are returned for further use in tests.
+
+    Examples
+    --------
+    To check that an Iteration tree has the following structure:
+
+    .. code-block:: python
+
+        for t
+            for x
+                for y
+            for f
+                for y
+
+    we call(Note: `time` mapped to `t`):
+
+    .. code-block:: python
+
+        trees, iters = assert_structure(op, ['t,x,y', 't,f,y'], 't,x,y,f,y')`
+    """
+    mapper = {'time': 't'}
+    trees = retrieve_iteration_tree(operator)
+    iters = FindNodes(Iteration).visit(operator)
```

Comment:
this is not needed unless `exp_iters is not None`

Reply:
There are cases where the structure is simple, so exp_iters is not tested as redundant. There are also tests that use Iterations later. So it is needed.